CHANGELOG.d/fix_avoids redundant substitution.md

```diff
@@ -0,0 +1 @@
+* Avoids unnecessary substitution checks when running `unifyTypes`. By only checking substitutions for unknown types, this leads to a 1.5-2x speedup for heavily constrained files.
```

Comment:
Just a thought. The "By only checking..." sentence should be a comment describe the first sentence.

```markdown
* Avoids unnecessary substitution checks when running `unifyType`.

  By only checking substitution for unknown types, this leads...
```